### DISCUSSION TITLE
Google Oauth

### DISCUSSION THREAD

**Discussion by Daniel-Boluda** [NginxProxyManager/nginx-proxy-manager#3326] [General]
Hello, 

First of all i need to say that i really love your project, its super nice user interface to set and configure nginx.

I'm trying to configure google auth to protect some of the applications i have behind Proxy Hosts, i have my container for google auth, looking as follows, but i'm having some problems to set it in Nginx-proxy-manager, i imagine something under the "advanced" section needs to be done, but i dont find any doc or example on how to do it.

  # Google Auth Service
  auth-service:
    image: quay.io/oauth2-proxy/oauth2-proxy
    environment:
      - OAUTH2_PROXY_CLIENT_ID=blablablabla.apps.googleusercontent.com
      - OAUTH2_PROXY_CLIENT_SECRET=blablablabla
      - OAUTH2_PROXY_COOKIE_SECRET=blablablabla
      - OAUTH2_PROXY_REDIRECT_URL=https://mydomain/oauth2/callback
      - OAUTH2_PROXY_AUTHENTICATED_EMAILS_FILE=/etc/oauth2_proxy/authenticated-emails.txt
    volumes:
      - /media/Config/nginx-proxy-manager/oauth2-proxy:/etc/oauth2_proxy
    ports:
      - "4180:4180"
    restart: always

**Comment by Daniel-Boluda** [comment]
someone has any ideas / did this before?

**Reply by nikitoyd** [reply]
Take a look below my config for Vhost I want to put behind google oauth

I have following config for '/' location
```
auth_request     /oauth2/auth;
error_page 401 = /oauth2/sign_in;
```

and following in advanced section
```
location /oauth2/ {
    proxy_pass       http://oauth2:4180;
    proxy_set_header Host                    $host; 
    proxy_set_header X-Real-IP               $remote_addr;
    proxy_set_header X-Scheme                $scheme;
    proxy_set_header X-Auth-Request-Redirect $scheme://$host$request_uri;
  }
  location = /oauth2/auth {
    proxy_pass       http://oauth2:4180;
    proxy_set_header Host             $host;
    proxy_set_header X-Real-IP        $remote_addr;
    proxy_set_header X-Scheme         $scheme;
    proxy_set_header Content-Length   "";
    proxy_pass_request_body           off;
  }
```

The cons of approach - I didn't find how to set "allowed list" of emails separately per application. I don't wan't to keep few oauth2 containers. TBH I didn't search a lot for info. If you know how to do so - please share it.
